xbmc/windowing/GraphicContext.cpp

```diff
@@ -333,13 +333,20 @@ void CGraphicContext::SetFullScreenVideo(bool bOnOff)
           bTriggerUpdateRes = true;
       }
     }
-
+    
+    bool allowResolutionChangeOnStop = CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE) != ADJUST_REFRESHRATE_ON_START;
+    RESOLUTION targetResolutionOnStop = RES_DESKTOP;
```

Comment:
RES_DESKTOP is not necessarily the resolution user has, when coming in here. That needs to be adjusted. the the actual resolution.

E.g. if user has Xserver in default resolution 3840x2160@60 and changes in kodi to 1920x1080@60 - then starts a movie. With targetResolutionOnStop = RES_DESKTOP it will switch to 4k@60 which is not right.

It should be: CDisplaySettings::GetInstance().GetCurrentResolution())

the else if in combination with the first if causes this wrong behaviour.

Reply:
Damn - i overlooked that bTriggerUpdateRes flag obviously.

Reply:
added check for bTriggerUpdateRes which should restore former behavior for all other settings but the new one.